### DISCUSSION TITLE
How to quickly write/read a geopandas object?

### DISCUSSION THREAD

**Discussion by kendonB** [geopandas/geopandas#2898] [Q&A] [answered]
I am looking for a solution to quickly serialise and write then read a geopandas object. 

Something functionally similar to R's https://github.com/traversc/qs would be ideal. What's out there for this task?

**Comment by martinfleis** [comment, marked as answer]
If you just want to dump the object that you have in memory to disk and then read it back, you can use either `pickle` or `joblib` packages.

```py
import pickle

# dump to disk
with open("gdf.pickle", "wb") as f:
    pickle.dump(gdf, f)

# read back to memory
with open("gdf.pickle", "rb") as f:
    gdf2 = pickle.load(f)
```

Same with `joblib`, which gives you a couple more options like compression.

```py
import joblib

with open("gdf.joblib.compressed", "wb") as f:
    joblib.dump(gdf, f, compress=True)

with open("gdf.joblib.compressed", "rb") as f:
    gdf3 = joblib.load(f)
```

An alternative is to save the GeoDataFrame to GeoParquet (`.to_parquet`) which tends to be performant and is interoperable.

**Comment by kendonB** [comment]
Thanks, this is helpful. The parallelised compression algorithms in `qs` are what makes it so fast and it doesn't look like `to_parquet` has those, but I will try it out. Thanks so much for the quick response.
